Code/Framework/AzCore/AzCore/Jobs/Internal/JobManagerWorkStealing.h

```diff
@@ -45,6 +52,40 @@ namespace AZ
             AZStd::deque<Job*> m_queue;
             LockType m_lock;
         };
+#else
+        // The Job Queue is a lock free 4-priority queue. Its basic operation is as follows:
+        // Each priority level is associated with a different queue, corresponding to the maximum size of a uint16_t.
+        // Each queue is implemented as a ring buffer, and a 64 bit atomic maintains the following state per queue:
+        // - offset to the "head" of the ring, from where we acquire elements
+        // - offset to the "tail" of the ring, which tracks where new elements should be enqueued
+        // - offset to a tail reservation index, which is used to reserve a slot to enqueue elements
+        class JobQueue final
+        {
+        public:
+            // Preallocating upfront allows us to reserve slots to insert jobs without locks.
+            // Each thread allocated by the job manager consumes ~2 MB.
+            constexpr static uint16_t MAX_QUEUE_SIZE = 0xffff;
```

Comment:
Nitpick: Prefer `AZStd::numeric_limits<uint16_t>::max()` as it's more readable.

Reply:
I agree but I'm actually going to do a pass to reduce as many std headers from this file (and others in the dependency tree of Job.h) as possible. AZStd::numeric_limits maps to the <limits> header it looks like.